### PR TITLE
Add initial schema

### PR DESCRIPTION
**Pull request by mkerry-moj**
* Add flyway
* Add spring data

### CODE REVIEW

File: src/main/resources/banner.txt [hunk 1]
```diff
@@ -1,13 +1,2 @@
...
- |  |___ |\/| |__] |    |__|  |  |___ 
- |  |___ |  | |    |___ |  |  |  |___ 
-                                      
-_  _ ____ ___ _    _ _  _ 
-|_/  |  |  |  |    | |\ | 
-| \_ |__|  |  |___ | | \| 
-                          
-Please change me by generating your own ASCII art and placing in banner.txt
+|__|  |\/| |__) /__` /__`    /  \ |\ | |__     |__) |     /\  |\ |     /\  |__) |
+|  |  |  | |    .__/ .__/    \__/ | \| |___    |    |___ /~~\ | \|    /~~\ |    |
```
Comment (tomhalson) [inline src/main/resources/banner.txt:2]: love this 😁

File: src/main/resources/db/migration/postgres/V1_1__initial.sql [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    type          smallint                    not null
+);
+
+create table objective
+(
+    id                     uuid primary key,
+    reference              uuid                        not null,
+    created_at             timestamp(3) with time zone not null,
+    created_by             varchar(50)                 not null,
+    updated_at             timestamp(3) with time zone not null,
```
Comment (tomhalson) [inline src/main/resources/db/migration/postgres/V1_1__initial.sql:20]: should updated_at and updated_by allow nulls?

Comment (mkerry-moj) [reply src/main/resources/db/migration/postgres/V1_1__initial.sql:20]: I figured they would default to created_at/created_by but honestly I don't remember what all the jpa stuff does by default

**Review by tomhalson — APPROVED** [review]

File: src/main/resources/banner.txt (hunk 1 above)
Comment (mkerry-moj) [reply src/main/resources/banner.txt:2]: I just realized I got the acronym wrong again! 🤦

**Pull request merged**